GenerateAndDownloadSurveyLinksExternalModule.php

```diff
@@ -20,32 +20,32 @@ class GenerateAndDownloadSurveyLinksExternalModule extends AbstractExternalModul
      * example of this same concept.
      */
     function generateDocumentSurveyLinksCron($cronAttributes){
-//        $hourRange = 6;
-//        if(date('G') > $hourRange){
-//            // Only perform actions between 12am and 6am.
-//            return;
-//        }
-//        $thisCron = $cronAttributes['cron_name'];
-//
-//        ## Old way of checking cron status, check to prevent re-sending on upgrade
-//        $lastRunSettingName = 'last-cron-run-time';
-//        $lastRun = (int)$this->getSystemSetting($lastRunSettingName);
-//        $hoursSinceLastRun = (time()-$lastRun)/60/60;
-//        if($hoursSinceLastRun < $hourRange){
-//            // We're already run recently
-//            return;
-//        }
-//
-//        $lastRunCronSettingName = 'last-cron-run-time-'.$thisCron;
-//        $lastRunThisCron = (int)$this->getSystemSetting($lastRunCronSettingName);
-//        $hoursSinceLastRun = (time()-$lastRunThisCron)/60/60;
-//        if($hoursSinceLastRun < $hourRange){
-//            // We're already run this cron recently
-//            return;
-//        }
-//
-//        ## Immediately log starting in case a second process spawns for this cron
-//        $this->setSystemSetting($lastRunCronSettingName, time());
+        $hourRange = 6;
+        if(date('G') > $hourRange){
+            // Only perform actions between 12am and 6am.
+            return;
+        }
+        $thisCron = $cronAttributes['cron_name'];
+
+        ## Old way of checking cron status, check to prevent re-sending on upgrade
+        $lastRunSettingName = 'last-cron-run-time';
+        $lastRun = (int)$this->getSystemSetting($lastRunSettingName);
+        $hoursSinceLastRun = (time()-$lastRun)/60/60;
+        if($hoursSinceLastRun < $hourRange){
+            // We're already run recently
+            return;
+        }
+
+        $lastRunCronSettingName = 'last-cron-run-time-'.$thisCron;
+        $lastRunThisCron = (int)$this->getSystemSetting($lastRunCronSettingName);
+        $hoursSinceLastRun = (time()-$lastRunThisCron)/60/60;
+        if($hoursSinceLastRun < $hourRange){
+            // We're already run this cron recently
+            return;
+        }
```

Comment:
For readability, I would consider putting these into private methods, given that these variables are not used later in the function.   They seem to be checks before you get to the actual substance of the function.  This is very minor, but you could also store `last-cron-run-time` as a property and append `" - . $thisCron;` instead of redefining it.